specification/redis/Cache.Management/models.tsp

```diff
@@ -0,0 +1,1052 @@
+  Deleted: "Deleted",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  Canceled: "Canceled",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  Failed: "Failed",
+}
+
+/**
+ * The Redis access key to regenerate.
+ */
+#suppress "@azure-tools/typespec-azure-core/no-enum" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+enum RedisKeyType {
+  Primary,
+  Secondary,
+}
+
+/**
+ * Day of the week when a cache can be patched.
+ */
+#suppress "@azure-tools/typespec-azure-core/no-enum" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+enum DayOfWeek {
+  Monday,
+  Tuesday,
+  Wednesday,
+  Thursday,
+  Friday,
+  Saturday,
+  Sunday,
+  Everyday,
+  Weekend,
+}
+
+/**
+ * Role of the linked server.
+ */
+#suppress "@azure-tools/typespec-azure-core/no-enum" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+enum ReplicationRole {
+  Primary,
+  Secondary,
+}
+
+/**
+ * REST API operation
+ */
+model Operation {
+  /**
+   * Operation name: {provider}/{resource}/{operation}
+   */
+  name?: string;
+
+  /**
+   * The object that describes the operation.
+   */
+  display?: OperationDisplay;
+}
+
+/**
+ * The object that describes the operation.
+ */
+model OperationDisplay {
+  /**
+   * Friendly name of the resource provider
+   */
+  provider?: string;
+
+  /**
+   * Operation type: read, write, delete, listKeys/action, etc.
+   */
+  operation?: string;
+
+  /**
+   * Resource type on which the operation is performed.
+   */
+  resource?: string;
+
+  /**
+   * Friendly name of the operation
+   */
+  description?: string;
+}
+
+/**
+ * Parameters body to pass for resource name availability check.
+ */
+model CheckNameAvailabilityParameters {
+  /**
+   * Resource name.
+   */
+  name: string;
+
+  /**
+   * Resource type. The only legal value of this property for checking redis cache name availability is 'Microsoft.Cache/redis'.
+   */
+  type: string;
+}
+
+/**
+ * The response of listUpgradeNotifications.
+ */
+model NotificationListResponse is Azure.Core.Page<UpgradeNotification>;
+
+/**
+ * Properties of upgrade notification.
+ */
+model UpgradeNotification {
+  /**
+   * Name of upgrade notification.
+   */
+  @visibility(Lifecycle.Read)
+  name?: string;
+
+  /**
+   * Timestamp when upgrade notification occurred.
+   */
+  @visibility(Lifecycle.Read)
+  // FIXME: (utcDateTime) Please double check that this is the correct type for your scenario.
+  timestamp?: utcDateTime;
+
+  /**
+   * Details about this upgrade notification
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-no-record" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @visibility(Lifecycle.Read)
+  upsellNotification?: Record<string>;
```

Comment:
For some reason I'm not finding upsellNotification in the codebase, I wonder if we stopped returning it.